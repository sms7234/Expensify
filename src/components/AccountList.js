import React from  'react';
import {connect} from 'react-redux';
import AccountListItem from './AccountListItem';
import selectAccounts from '../selectors/accounts';


export const AccountList = (props) => (
  <div className="content-container">
    <div className="list-header">
      <div className="show-for-mobile">Accounts</div>
      <div className="show-for-desktop">Accounts</div>
      <div className="show-for-desktop">Description</div>
    </div>
    <div className="list-body">
    {
      props.accounts.length === 0 ? (
        <div className="list-item--message">
          <span>No accounts to show</span>
        </div>
      ): (
        props.accounts.map((item) => {return <AccountListItem key={item.id}{...item}/>;})
      )
    }
    </div>
  </div>
);

const mapStateToProps = (state) => {
  return {
    accounts: selectAccounts(state.accounts, state.filters)
  };
};

export default connect(mapStateToProps)(AccountList);
