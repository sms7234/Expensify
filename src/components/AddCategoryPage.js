import React from 'react';
import CategoryForm from './CategoryForm';
import {connect} from 'react-redux';
import {startAddCategory} from '../actions/categories';

export class AddCategoryPage extends React.Component{
  onSubmit=(category) => {
    this.props.startAddCategory(category);
    this.props.history.push('/categories');
  };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title"> Add Category </h1>
          </div>
        </div>
        <div className="content-container">
          <CategoryForm onSubmit={this.onSubmit} />
        </div>
      </div>
    );
  };
};

const mapDispatchToProps = (dispatch) => ({
  startAddCategory:(category) => dispatch(startAddCategory(category))
});

export default connect(undefined, mapDispatchToProps)(AddCategoryPage);
