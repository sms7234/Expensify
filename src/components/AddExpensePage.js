import React from 'react';
import ExpenseForm from './ExpenseForm';
import { connect} from 'react-redux';
import startAddExpense from '../actions/expenses';

export class AddExpensePage extends React.Component{
  onSubmit = (expense) => {
    this.props.startAddExpense(expense);
    this.props.history.push('/expenses');
  };
  render(){
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title"> Add Expense</h1>
          </div>
        </div>
        <div className="content-container">
          <ExpenseForm
            onSubmit = {this.onSubmit}
            categoryList = {this.props.categoryList}
          />
        </div>
      </div>
    );
  }
};

const mapStateToProps=(state) => {
  return {
    categoryList: state.categories.map((item)=>{return({label: item.category, value: item.category})})
  };
};

const mapDispatchToProps = (dispatch) => ({
  startAddExpense: (expense) => dispatch(startAddExpense(expense))
});

export default connect(mapStateToProps, mapDispatchToProps)(AddExpensePage);
