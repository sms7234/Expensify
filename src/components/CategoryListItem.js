import React from 'react';
import {Link} from 'react-router-dom';

const CategoryListItem=({dispatch, id, category, description}) =>(
  <Link className="list-item" to={`/add`}>
    <div>
      <h3 className="list-item__title">{category}</h3>
    </div>
    <h3 className="list-item__data">{description}</h3>
  </Link>
)

export default CategoryListItem;
