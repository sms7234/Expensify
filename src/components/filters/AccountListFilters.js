import React from 'react';
import {connect} from 'react-redux';
import {setAccountFilter} from '../../actions/filters';

export class AccountListFilters extends React.Component {
  onTextChange = (e) => {
    this.props.setAccountFilter(e.target.value);
  };
  render () {
    return (
      <div className="content-container">
        <div className="input-group">
          <div className="input-group__item">
            <input
              type="text"
              className="text-input"
              placeholder="search accounts"
              value={this.props.filters.account}
              onChange={this.onTextChange}
            />
          </div>
        </div>
      </div>
    );
  }
};

const mapStateToProps = (state) => {
  return {
    filters: state.filters
  };
};

const mapDispatchToProps = (dispatch) => ({
  setAccountFilter: (text) => dispatch(setAccountFilter(text)),
});

export default connect(mapStateToProps, mapDispatchToProps)(AccountListFilters);
