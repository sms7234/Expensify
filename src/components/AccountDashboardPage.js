import React from 'react';
import AccountList from './AccountList';
import AccountSummary from './AccountSummary';
import AccountListFilters from './AccountListFilters';

const AccountDashboardPage = () => (
  <div>
    <AccountSummary />
    <AccountListFilters />
    <AccountList />
  </div>
)

export default AccountDashboardPage;
