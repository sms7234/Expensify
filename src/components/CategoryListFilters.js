import React from 'react';
import {connect} from 'react-redux';
import {setTextFilter} from '../actions/filters';

export class CategoryListFilters extends React.Component {
  onTextChange = (e) => {
    this.props.setTextFilter(e.target.value);
  };
  render () {
    return (
      <div className="content-container">
        <div className="input-group">
          <div className="input-group__item">
            <input
              type="text"
              className="text-input"
              placeholder="search categories"
              value={this.props.filters.text}
              onChange={this.onTextChange}
            />
          </div>
        </div>
      </div>
    );
  }
};

const mapStateToProps = (state) => {
  return {
    filters: state.filters
  };
};

const mapDispatchToProps = (dispatch) => ({
  setTextFilter: (text) => dispatch(setTextFilter(text)),
});

export default connect(mapStateToProps, mapDispatchToProps)(CategoryListFilters);
