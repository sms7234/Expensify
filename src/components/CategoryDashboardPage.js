import React from 'react';
import CategoryList from './CategoryList';
import CategoryListFilters from './CategoryListFilters';
import CategorySummary from './CategorySummary';

const CategoryDashboardPage = () => (
  <div>
    <CategorySummary />
    <CategoryListFilters />
    <CategoryList />
  </div>
)

export default CategoryDashboardPage;
