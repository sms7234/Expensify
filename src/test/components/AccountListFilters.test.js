import React from 'react';
import {shallow} from 'enzyme';
import {AccountListFilters} from '../../components/AccountListFilters';

let setAccountFilter, wrapper;

const filters = {
  account: ''
};

beforeEach(()=> {
  setAccountFilter = jest.fn();
  wrapper=shallow(
    <AccountListFilters
      filters={filters}
      setAccountFilter={setAccountFilter} />
  );
});

test('should render AccountListFilters correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should handle text input', () => {
  const value="texxxxxt";
  wrapper.find('input').simulate('change', {
    target: {value}
  });
  expect(setAccountFilter).toHaveBeenLastCalledWith(value);
});
