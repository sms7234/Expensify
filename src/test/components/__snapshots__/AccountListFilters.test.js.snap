// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AccountListFilters correctly 1`] = `
<div
  className="content-container"
>
  <div
    className="input-group"
  >
    <div
      className="input-group__item"
    >
      <input
        className="text-input"
        onChange={[Function]}
        placeholder="search accounts"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
